protobuf/src/main/scala/magnolify/protobuf/ProtobufType.scala

```diff
@@ -29,26 +29,26 @@ import scala.annotation.implicitNotFound
 import scala.language.experimental.macros
 import scala.reflect.ClassTag
 
-sealed trait ProtobufType[T, ParentMsgT <: Message] extends Converter[T, ParentMsgT, ParentMsgT] {
-  def apply(r: ParentMsgT): T = from(r)
-  def apply(t: T): ParentMsgT = to(t)
+sealed trait ProtobufType[T, MsgT <: Message] extends Converter[T, MsgT, MsgT] {
+  def apply(r: MsgT): T = from(r)
+  def apply(t: T): MsgT = to(t)
 }
 
 object ProtobufType {
-  implicit def apply[T, ParentMsgT <: Message: ClassTag](
-    implicit f: ProtobufField.Record[T]
-  ): ProtobufType[T, ParentMsgT] =
-    new ProtobufType[T, ParentMsgT] {
-      override def from(v: ParentMsgT): T = f.from(v)
-      override def to(v: T): ParentMsgT =
+  implicit def apply[T, MsgT <: Message](
+    implicit f: ProtobufField.Record[T], ct: ClassTag[MsgT]
```

Comment:
I tend to add `implicit` argument if it needs to be accessed via `implicitly` in the method anyway.